### DISCUSSION TITLE
FAST with FastPathology pipeline, export with fast.ImageExporter

### DISCUSSION THREAD

**Discussion by mhoibo** [FAST-Imaging/FAST#199] [Q&A] [answered]
Hi, I have tried to use a pipeline from the fast pathology datahub (breast epithelium segmentation) with png-images from a public dataset that are not pyramidal. I edited the pipeline instead of using it out of the box, and used ImageImporter and ImageExporter: 

    model = model_onnx = '/home/user/FAST/datahub/breast-epithelium-segmentation-model/' \
                 'breast_epithelium_segmentation_single_output.onnx'

    importer = fast.ImageImporter.create(path_)  # path to png image

    tissue_seg = fast.TissueSegmentation.create(threshold=70).connect(importer)

    generator = fast.PatchGenerator.create(1024, 1024, overlapPercent=0.3,
                                           maskThreshold=0.1).connect(0, importer).connect(1, tissue_seg)

    network = fast.SegmentationNetwork.create(model_, scaleFactor=1./255.). \
        connect(generator)

    stitcher = fast.PatchStitcher.create().connect(network)

    finished = fast.RunUntilFinished.create().connect(stitcher)

    # save segmentation on disk
    os.makedirs(seg_path, exist_ok=True)
    fast.ImageExporter.create(file_path) \
        .connect(finished).run()
 
But i get an error on the ImageExporter(): 

     RuntimeError: Bad cast. Tried to cast from type Image to ImagePyramid

Do you know why the export is problematic, and if there is a possible solution to it? I have used inputs to this model before which were not vsis, but images made with fast.Image.createFromArray(). If I convert to numpy, then create a fast image with it I get the exact same error. Other exporters also yield the same error.

**Comment by smistad** [comment, marked as answer]
Hi. I believe this error is coming from TissueSegmentation which expects an ImagePyramid. You can verify that by removing it.

I will add a small fix to TissueSegmentation which makes it support regular images as well.

**Reply by mhoibo** [reply]
Hi, yes, that worked. Thank you, @smistad

**Reply by smistad** [reply]
Here is a fix for TissueSegmentation https://github.com/smistad/FAST/commit/ca375a659cfde25cc174cfcf37447b64d24e5244 @mhoibo 
Now you can send a regular Image object to TissueSegmentation.
